### PR TITLE
feat: add italian and arabic categories, refactor categories name

### LINKED ISSUE #536: Add "Italian" language category
Probably a good idea to add it as a migration (with the other "preset" categories as well)

### PR DESCRIPTION
**Pull request by pyphilia**
close #536

### CODE REVIEW

File: src/migrations/1692624998160-category-seed.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        );
+    `);
+
+    // update categories for discipline
+    await update('Arts', 'arts');
+    await update('Language', 'language');
+    await update('Math', 'mathematics');
+    await update('Literature', 'literature');
+    await update('Natural Science', 'natural-science');
+    await update('Social Science', 'social-science');
```
Comment (spaenleh) [inline src/migrations/1692624998160-category-seed.ts:36]: Should we add `informatics` ? And `physical-education` ?

File: src/migrations/1692624998160-category-seed.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    await queryRunner.query(`INSERT INTO category (name, type) VALUES (
+      'collection',
+      'resource-type'
+      );
+    `);
+    await queryRunner.query(`INSERT INTO category (name, type) VALUES (
+      'template',
+      'resource-type'
+      );
+    `);
```
Comment (spaenleh) [inline src/migrations/1692624998160-category-seed.ts:61]: Question: We will have to implement something in the frontend to set these right ?

File: src/migrations/1692624998160-category-seed.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    // update categories for discipline
+    await update('Arts', 'arts');
+    await update('Language', 'language');
+    await update('Math', 'mathematics');
+    await update('Literature', 'literature');
+    await update('Natural Science', 'natural-science');
+    await update('Social Science', 'social-science');
+
+    // update categories for level
+    await update('Kindergarden', 'kindergarten');
```
Comment (spaenleh) [inline src/migrations/1692624998160-category-seed.ts:39]: Is the change from `kindergarden` to `kindergarten` intentional ?

File: src/migrations/1692624998160-category-seed.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    await queryRunner.query(`UPDATE FROM category SET name='Literature' WHERE name = 'literature'`);
+    await queryRunner.query(
+      `UPDATE FROM category SET name='Natural Science' WHERE name = 'natural-science'`,
+    );
+    await queryRunner.query(
+      `UPDATE FROM category SET name='Social Science' WHERE name = 'social-science'`,
+    );
+
+    await queryRunner.query(
+      `UPDATE FROM category SET name='Kindergarten' WHERE name = 'kindergarten'`,
```
Comment (spaenleh) [inline src/migrations/1692624998160-category-seed.ts:84]: `Kindergarden` ?

File: test/migrations/migrations1692624998160/migrations.test.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    const categories = await app.db.query('select * from category');
+    const names = categories.map((c) => c.name);
+    expect(names).toContain('arts');
+    expect(names).toContain('language');
+    expect(names).toContain('mathematics');
+    expect(names).toContain('literature');
+    expect(names).toContain('natural-science');
+    expect(names).toContain('social-science');
+    expect(names).toContain('kindergarten');
```
Comment (spaenleh) [inline test/migrations/migrations1692624998160/migrations.test.ts:48]: If using the `kindergarden` we also need to change here.

**Review by spaenleh — APPROVED** [review]
See my comments

File: src/migrations/1692624998160-category-seed.ts (hunk 2 above)
Comment (pyphilia) [reply src/migrations/1692624998160-category-seed.ts:61]: At least add it in the enum. Then I think Builder will automatically update, but not sure. I don't know for library.

File: src/migrations/1692624998160-category-seed.ts (hunk 3 above)
Comment (pyphilia) [reply src/migrations/1692624998160-category-seed.ts:39]: The `t` one is the correct one!

**Review by spaenleh — APPROVED** [review]

**Review by dialexo — APPROVED** [review]
LGTM 🌈 consider writing the down migration in the same or inverse order from the up one which would make it more readable 👍

**Pull request merged**